src/dataflow-types/src/client/controller.rs

```diff
@@ -90,8 +90,11 @@ where
 
         // Add replicas backing that instance.
         match config {
-            InstanceConfig::Virtual => {
-                let client = self.virtual_compute_host.create_instance(instance);
+            InstanceConfig::Local => {
+                let client = self
+                    .local_compute
+                    .take()
+                    .expect("cannot create more than one local compute instance");
```

Comment:
I can't really tell what is going on here. We are popping out the existing local compute, and then installing it elsewhere? Is `local_compute` just a temporary stash before installing it?

Reply:
Yeah, exactly. I added a comment to explain. It's ugly, but I tried a few other approaches and none of them panned out.